### PR TITLE
conformance: Adds Windows considerations for tests

### PR DESCRIPTION
**Pull request by bclau**
There are a number of tests labeled as ``[LinuxOnly]``. This commit documents those reasons, so the community can take them
into considerations when labeling such tests, or promoting other tests to Conformance.

Fixes: https://github.com/kubernetes/kubernetes/issues/75741

### CODE REVIEW

File: contributors/devel/sig-architecture/conformance-tests.md [hunk 1]
```diff
@@ -154,6 +154,43 @@ To promote a test to the conformance test suite, open a PR as follows:
...
+  fail.
+- Pod ``HostNetwork=true`` set. Is not supported on Windows, and if set, the Pod will
+  fail to spawn.
+- Cannot resolve PQDNs (Partially Qualified Domain Names). At the moment, Windows Containers are
+  configured with a single DNS suffix (e.g.: ``lish.svc.cluster.local``). The containers are
+  only able to resolve FQDNs, or names within namespace (e.g.: ``foo.lish.cluster.local``). Anything
+  in between it cannot (e.g.: ``foo.lish``), and tests relying on PQDNs will fail.
+- Running Linux specific commands. Most Windows images used in testing are based on busybox, and thus,
+  quite a few Linux commands are acceptable. But, due to the differences between Linux and Windows, not
+  all commands and command options are included in Windows busybox binary. The most concerning cases are:
```
Comment (bclau) [inline contributors/devel/sig-architecture/conformance-tests.md]: The commands found in the Windows busybox binary are:

    Currently defined functions:
        [, [[, ar, ash, awk, base64, basename, bash, bunzip2, bzcat, bzip2, cal, cat, catv, chmod, cksum, clear, cmp, comm, cp, cpio,
        cut, date, dc, dd, df, diff, dirname, dos2unix, dpkg-deb, du, echo, ed, egrep, env, expand, expr, false, fgrep, find, fold,
        ftpget, ftpput, getopt, grep, groups, gunzip, gzip, hd, head, hexdump, id, ipcalc, kill, killall, less, ln, logname, ls,
        lzcat, lzma, lzop, lzopcat, man, md5sum, mkdir, mktemp, mv, nc, od, patch, pgrep, pidof, printenv, printf, ps, pwd, rev, rm,
        rmdir, rpm2cpio, sed, seq, sh, sha1sum, sha256sum, sha3sum, sha512sum, shuf, sleep, sort, split, stat, strings, sum, tac,
        tail, tar, tee, test, touch, tr, true, truncate, uname, uncompress, unexpand, uniq, unix2dos, unlink, unlzma, unlzop, unxz,
        unzip, usleep, uudecode, uuencode, vi, wc, wget, which, whoami, whois, xargs, xz, xzcat, yes, zcat

I wonder if it's worth adding them here.

**Comment by timothysc** [comment]
@bclau @PatrickLang which PR do you want to merge?

**Comment by timothysc** [comment]
It looks like you merged this content into #3521, do you intent to close this issue?

**Comment by bclau** [comment]
Yep, the contents of this PR has been included in Patrick's PR. Closing this.

**Pull request closed without merging**